src/posthog-surveys.ts

```diff
@@ -261,38 +291,50 @@ export class PostHogSurveys {
         const existingSurveys = this.instance.get_property(SURVEYS)
 
         if (!existingSurveys || forceReload) {
-            this.instance._send_request({
-                url: this.instance.requestRouter.endpointFor(
-                    'api',
-                    `/api/surveys/?token=${this.instance.config.token}`
-                ),
-                method: 'GET',
-                callback: (response) => {
-                    const statusCode = response.statusCode
-                    if (statusCode !== 200 || !response.json) {
-                        logger.error(`Surveys API could not be loaded, status: ${statusCode}`)
-                        return callback([])
-                    }
-                    const surveys = response.json.surveys || []
-
-                    const eventOrActionBasedSurveys = surveys.filter(
-                        (survey: Survey) =>
-                            (survey.conditions?.events &&
-                                survey.conditions?.events?.values &&
-                                survey.conditions?.events?.values?.length > 0) ||
-                            (survey.conditions?.actions &&
-                                survey.conditions?.actions?.values &&
-                                survey.conditions?.actions?.values?.length > 0)
-                    )
-
-                    if (eventOrActionBasedSurveys.length > 0) {
-                        this._surveyEventReceiver?.register(eventOrActionBasedSurveys)
-                    }
+            // Prevent concurrent API calls
+            if (this._isFetchingSurveys) {
+                return
+            }
 
-                    this.instance.persistence?.register({ [SURVEYS]: surveys })
-                    return callback(surveys)
-                },
-            })
+            try {
+                this._isFetchingSurveys = true
+                this.instance._send_request({
+                    url: this.instance.requestRouter.endpointFor(
+                        'api',
+                        `/api/surveys/?token=${this.instance.config.token}`
+                    ),
+                    method: 'GET',
+                    callback: (response) => {
```

Comment:
if there's no connection, is the `callback` still called so we can reset the `_isFetchingSurveys`?
or is it gonna throw and the `catch` will reset it?

Reply:
This needs to be tested, but I believe `callback` will still be called even if there's no connection

Reply:
agree, and this [test](https://github.com/PostHog/posthog-js/pull/1747/files#r1962233991) actually calls the callback in the test, so its not really testing what it would happen in case of a timeout/no connection